### PR TITLE
Fix http params handling discrepancy 

### LINKED ISSUE #909: Not throw 400 error when params is empty
Some APIs don't need any parameter. We should allow params is empty in this case.

Request:
```
{"method":"server_info","params":[]}
{"method":"server_info","params":[null]}
{"method":"server_info","params": ""}
{"method":"server_info","params": null}
{"method":"server_info","params": {}}
```
Clio returns 400 error
Rippled returns response

### PR DESCRIPTION
**Pull request by godexsoft**
Fixes #909

### CODE REVIEW

File: unittests/web/RPCServerHandlerTests.cpp [hunk 1]
```diff
@@ -549,7 +549,7 @@ TEST_F(WebRPCServerHandlerTest, HTTPParamsUnparseableNotArray)
     EXPECT_EQ(session->lastStatus, boost::beast::http::status::bad_request);
 }
 
-TEST_F(WebRPCServerHandlerTest, HTTPParamsUnparseableEmptyArray)
+TEST_F(WebRPCServerHandlerTest, HTTPParamsUnparseableArrayWithDigit)
```
Comment (godexsoft) [inline unittests/web/RPCServerHandlerTests.cpp:552]: These should probably be made into a parameterised test and we check `[]`, `null`, `[{}]`, `[""]`, `""`, `null` and `{}` as success cases, `1` and `[1]` as failures.

File: src/web/RPCServerHandler.h [hunk 2]
```diff
@@ -89,8 +89,8 @@ class RPCServerHandler
             auto req = boost::json::parse(request).as_object();
             LOG(perfLog_.debug()) << connection->tag() << "Adding to work queue";
 
-            if (not connection->upgraded and not req.contains("params"))
-                req["params"] = boost::json::array({boost::json::object{}});
+            if (not connection->upgraded && shouldReplaceParams(req))
```
Comment (cindyyan317) [inline src/web/RPCServerHandler.h]: small thing : "and" is better here, because "not" had been used here.

**Review by cindyyan317 — DISMISSED** [review]
LGTM 👍

File: src/web/RPCServerHandler.h (hunk 2 above)
Comment (godexsoft) [reply src/web/RPCServerHandler.h]: Agree!

**Pull request merged**